lib/Parser/RegexParser.cpp

```diff
@@ -2147,22 +2153,27 @@ namespace UnifiedRegex
             else if (nextChar == '\\')
             {
                 Node* returnedNode = ClassEscapePass1(&deferredCharNode, &deferredSetNode, previousWasASurrogate);
+                codePointToSet = pendingCodePoint;
 
                 if (returnedNode->tag == Node::MatchSet)
                 {
-                    codePointToSet = pendingCodePoint;
+                    if (unicodeFlagPresent && pendingRangeStart != INVALID_CODEPOINT)
+                    {
+                        Fail(JSERR_UnicodeRegExpRangeContainsCharClass);
```

Comment:
>Fail(JSERR_UnicodeRegExpRangeContainsCharClass); [](start = 24, length = 48)

as another comment mentioned, this is pass1 so usually syntax error would have happened in pass 0. Since pass 0 doesn't have information about the unicode flag, this is the only place we can check. Just add a comment explaining this.